### PR TITLE
Use simpler TimeSeries class and implement rereferencing

### PR DESCRIPTION
**Pull request by mivade**

### CODE REVIEW

File: cmlreaders/readers/eeg.py [hunk 1]
```diff
@@ -118,6 +123,13 @@ def read(self) -> np.ndarray:
             raise NotImplementedError("FIXME: we can only read all channels now")
 
         with h5py.File(self.filename, 'r') as hfile:
+            try:
+                self.rereferencing_possible = bool(hfile['monopolar_possible'][0])
```
Comment (zduey) [inline cmlreaders/readers/eeg.py]: Would it make sense to raise some sort of warning if this is true so the user knows that they won't be able to reference this particular dataset?

Comment (zduey) [reply cmlreaders/readers/eeg.py]: nvm, I see this will happen inside of as_timeseries

File: cmlreaders/timeseries.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            data = np.concatenate([s.data for s in series], axis=0)
+            epochs = list(np.concatenate([s.epochs for s in series]))
+            attrs = dict(ChainMap(*[s.attrs for s in series]))
+
+            return TimeSeries(data, samplerate, epochs,
+                              channels=series[0].channels,
+                              tstart=series[0].time[0],
+                              attrs=attrs)
+
+        elif dim == "time":
```
Comment (zduey) [inline cmlreaders/timeseries.py]: I'm assuming this is just WIP?

Comment (mivade) [reply cmlreaders/timeseries.py]: Oops.

**Pull request merged**